frontend/src/pages/notebookController/screens/server/useSpawnerNotebookModalState.ts

```diff
@@ -1,11 +1,34 @@
 import * as React from 'react';
 import { useHistory } from 'react-router-dom';
 import { NotebookControllerContext } from '../../NotebookControllerContext';
+import { FAST_POLL_INTERVAL } from '../../../../utilities/const';
+import { NotebookControllerContextProps } from '../../notebookControllerContextTypes';
 
-const useSpawnerNotebookModalState = (): [
-  startShown: boolean,
-  setStartShown: (newState: boolean) => void,
-] => {
+const useRefreshNotebookAndCleanup = (startShown: boolean) => {
+  const { requestNotebookRefresh } = React.useContext(NotebookControllerContext);
+
+  const methodRef =
+    React.useRef<NotebookControllerContextProps['requestNotebookRefresh']>(requestNotebookRefresh);
+  methodRef.current = requestNotebookRefresh; // we don't care about the ref, we just want the last one
```

Comment:
This is to get around the reference behind the refresh. It will refresh this reference every time the notebook is updated, so we want to avoid those... in actuality, we only need the last one, so the reference integrity does not matter.